### PR TITLE
feat(types): Added types in `useApiCall` hooks

### PR DESCRIPTION
**Pull request by Gmin2**
Completed the migration to typescript for all the API calls in pages .

### CODE REVIEW

File: website/src/types/index.ts [hunk 1]
```diff
@@ -15,3 +15,70 @@ limitations under the License.
...
+
+export interface ComponentsCpuTime {
+  vtgate: MacroDataValue;
+  vttablet: MacroDataValue;
+}
+
+export interface ComponentsMemStatsAllocBytes {
+  vtgate: MacroDataValue;
+  vttablet: MacroDataValue;
+}
```
Comment (frouioui) [inline website/src/types/index.ts:61]: It feels like these two types could be merged together into, for instance: `ComponentsStats`.

File: website/src/pages/SearchPage/SearchPage.tsx [hunk 2]
```diff
@@ -42,27 +45,35 @@ export default function SearchPage() {
...
+      )}
 
       {isSearchLoading && (
         <div className="flex my-10 justify-center items-center">
           <RingLoader loading={isSearchLoading} color="#E77002" size={300} />
         </div>
       )}
 
-      {!isSearchLoading && dataSearch && (
+      {!isSearchLoading && dataSearch && dataSearch.length > 0 && (
```
Comment (frouioui) [inline website/src/pages/SearchPage/SearchPage.tsx:58]: The `dataSearch.length > 0` is a good idea and fix. Let's actually duplicate this for the `/compare` page.

**Review by frouioui — COMMENTED** [review]
I think overall the changes proposed in this Pull Request look good to me, I left two nits. I see in the PR description you want to add type definition for the compare page, let's do this. Moreover, could you go over the FK and Daily pages? That way we will be using proper types in every page.

Only page left would be the query comparison page, but this can be left for later, the API for that page has to change a little bit too anyway.

Not approving this as you're planning on doing more work, but looks great!

**Comment by frouioui** [comment]
After merging #538 there are some conflicts on this branch.

**Comment by utnim2** [comment]
> After merging #538 there are some conflicts on this branch.

My end semester are going on will be resolving it after the week

**Comment by frouioui** [comment]
> My end exams semester are going on will be resolving it after the week

No worries at all, take your time and good luck.

**Comment by utnim2** [comment]
Ready to review @frouioui

File: website/src/pages/DailyPage/DailyPage.tsx [hunk 3]
```diff
@@ -22,41 +22,47 @@ import useApiCall from "../../utils/Hook";
...
+  } = useApiCall<MacroData>(`${import.meta.env.VITE_API_URL}daily?type=${benchmarkType}`);
 
   const navigate = useNavigate();
 
   useEffect(() => {
     navigate(`?type=${benchmarkType}`);
   }, [benchmarkType]);
 
-  const TPSData = [
+  const TPSData: { id: string; data: { x: string; y: number }[] }[] = [
```
Comment (frouioui) [inline website/src/pages/DailyPage/DailyPage.tsx:58]: I am thinking we should extract `{ x: string; y: number }` as a type, we are reusing this in a bunch of places.

**Comment by Gmin2** [comment]
@frouioui can we merge this soon, its types are very important to migrate other component

**Review by frouioui — APPROVED** [review]

**Pull request merged**